pdns/syncres.cc

```diff
@@ -713,6 +713,35 @@ bool SyncRes::isForwardOrAuth(const DNSName &qname) const {
   return iter != t_sstorage.domainmap->end() && (iter->second.isAuth() || !iter->second.shouldRecurse());
 }
 
+# if 0
+// Will be needed in the future
+static const char* timestamp(const struct timeval& tv, char* buf, size_t sz)
+{
+  const std::string s_timestampFormat = "%Y-%m-%dT%T";
+  struct tm tm;
+  size_t len = strftime(buf, sz, s_timestampFormat.c_str(), localtime_r(&tv.tv_sec, &tm));
+  if (len == 0) {
+    len = snprintf(buf, sz, "%lld", static_cast<long long>(tv.tv_sec));
+  }
+
+  if (sz > len) {
+    snprintf(buf + len, sz - len, ".%03ld", static_cast<long>(tv.tv_usec) / 1000);
+  }
+  return buf;
+}
+#endif
+
+static const char* timestamp(time_t t, char* buf, size_t sz)
+{
+  const std::string s_timestampFormat = "%Y-%m-%dT%T";
+  struct tm tm;
+  size_t len = strftime(buf, sz, s_timestampFormat.c_str(), localtime_r(&t, &tm));
+  if (len == 0) {
+    snprintf(buf, sz, "%lld", static_cast<long long>(t));
```

Comment:
`snprintf` should never fail as long as long as a long long fit in 25 characters, but how about actually checking the return value and setting `buf[0] = '\0';` if it does just to be sure?